2023-01-17 15:04:31,639 ERROR: Exception on /user/susan [GET] [in /Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask/app.py:1741]
Traceback (most recent call last):
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask_login/utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
  File "/Users/joannachen/Desktop/flask_proj/app/routes.py", line 76, in user
    return render_template('user.html',user=user,posts=posts.items,form=form)
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask/templating.py", line 147, in render_template
    return _render(app, template, context)
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask/templating.py", line 130, in _render
    rv = template.render(context)
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/Users/joannachen/Desktop/flask_proj/app/templates/user.html", line 1, in top-level template code
    {% extends 'base.html' %}
  File "/Users/joannachen/Desktop/flask_proj/app/templates/base.html", line 1, in top-level template code
    {% extends 'bootstrap/base.html' %}
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 1, in top-level template code
    {% block doc -%}
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 4, in block 'doc'
    {%- block html %}
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 20, in block 'html'
    {% block body -%}
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 23, in block 'body'
    {% block content -%}
  File "/Users/joannachen/Desktop/flask_proj/app/templates/base.html", line 43, in block 'content'
    {% block app_content %}{% endblock %}
  File "/Users/joannachen/Desktop/flask_proj/app/templates/user.html", line 8, in block 'app_content'
    <h1>{{ _('User') }}: {{ user.username }}</h1>
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/jinja2/utils.py", line 83, in from_obj
    if hasattr(obj, "jinja_pass_arg"):
jinja2.exceptions.UndefinedError: '_' is undefined
2023-01-17 15:05:16,924 INFO: Microblog startup [in /Users/joannachen/Desktop/flask_proj/app/__init__.py:49]
2023-01-17 15:05:19,165 ERROR: Exception on /user/susan [GET] [in /Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask/app.py:1741]
Traceback (most recent call last):
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask_login/utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
  File "/Users/joannachen/Desktop/flask_proj/app/routes.py", line 76, in user
    return render_template('user.html',user=user,posts=posts.items,form=form)
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask/templating.py", line 147, in render_template
    return _render(app, template, context)
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask/templating.py", line 130, in _render
    rv = template.render(context)
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/Users/joannachen/Desktop/flask_proj/app/templates/user.html", line 1, in top-level template code
    {% extends 'base.html' %}
  File "/Users/joannachen/Desktop/flask_proj/app/templates/base.html", line 1, in top-level template code
    {% extends 'bootstrap/base.html' %}
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 1, in top-level template code
    {% block doc -%}
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 4, in block 'doc'
    {%- block html %}
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 20, in block 'html'
    {% block body -%}
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 23, in block 'body'
    {% block content -%}
  File "/Users/joannachen/Desktop/flask_proj/app/templates/base.html", line 43, in block 'content'
    {% block app_content %}{% endblock %}
  File "/Users/joannachen/Desktop/flask_proj/app/templates/user.html", line 8, in block 'app_content'
    <h1>{{ _('User') }}: {{ user.username }}</h1>
  File "/Users/joannachen/Desktop/flask_proj/venv/lib/python3.10/site-packages/jinja2/utils.py", line 83, in from_obj
    if hasattr(obj, "jinja_pass_arg"):
jinja2.exceptions.UndefinedError: '_' is undefined
2023-01-17 15:06:04,682 INFO: Microblog startup [in /Users/joannachen/Desktop/flask_proj/app/__init__.py:49]
